python/hyperon/exts/snet_io/snet_gate.py

```diff
@@ -137,9 +139,105 @@ def get_service_messages(self):
     def get_operation_atom(self):
         return OperationAtom(self.service_details[1], self)
 
+    def __pretty_print__(self, input_str):
+        list_of_functions = ["if", "match", "let", "let*", "unify",
```

Comment:
what is special about these functions?

Reply:
Those are functions which could have several inner brackets inside their body. Probably I could forget some functions.

Reply:
What's about just nested expressions like `(* (- $x $y) (- $x $y))`? What is the substantial difference?

Reply:
I suspect that it is problematic to detect nested brackets in your approach, because you try to reformat a textual representation, so you need to re-parse it. If you'd started with atoms, it could be simpler...